llm/throughput/run_all_configs.py

```diff
@@ -0,0 +1,306 @@
+    gpu_types = set()
+    for c in clusters:
+        for g in get_cluster_gpu_types(c):
+            gpu_types.add(g)
+    return gpu_types
+
+
+def get_gpu_nums(clusters, gpu_types):
+    max_gpus_per_run = 1
+    for c in clusters:
+        for gpu_info in CLUSTER_INFO[c]:
+            if gpu_info[0] in gpu_types:
+                max_gpus_per_run = max(max_gpus_per_run, gpu_info[1])
+    
+    gpu_nums = [1]
+    while gpu_nums[-1] < max_gpus_per_run:
+        gpu_nums += [2 * gpu_nums[-1]]
+
+    return gpu_nums
+
+
+def get_valid_gpu_lim(cluster, gpu_type):
+    for gpu_info in CLUSTER_INFO[cluster]:
+        if gpu_info[0] == gpu_type:
+            return gpu_info[1]
+    raise ValueError
+
+
+def mod_parameters(
+    parameters,
+    max_seq_len,
+    global_train_batch_size,
+    precision,
+    fsdp_config_mixed_precision='DEFAULT',
+    run_name='',
+    streaming_data=False,
+    max_duration='12ba',
+    eval_interval='500ba',
+    microbatch_size=None,
+    wandb=True,
+):
+    if run_name:
+        parameters['run_name'] = run_name
+    if streaming_data:
+        parameters['data_remote'] = "s3://mosaicml-internal-dataset-c4/mds/2"
+        parameters['train_loader']['dataset']['remote'] = parameters['data_remote']
+        parameters['eval_loader']['dataset']['remote'] = parameters['data_remote']
+        
+        parameters['data_local'] = "/tmp/c4"
+        parameters['train_loader']['dataset']['local'] = parameters['data_local']
+        parameters['eval_loader']['dataset']['local'] = parameters['data_local']
+    # set max_seq_len
+    parameters['max_seq_len'] = max_seq_len
+    parameters['model']['max_seq_len'] = max_seq_len
+    parameters['tokenizer']['args']['max_seq_len'] = max_seq_len
+    parameters['train_loader']['dataset']['max_seq_len'] = max_seq_len
+    parameters['eval_loader']['dataset']['max_seq_len'] = max_seq_len
+
+    parameters['global_train_batch_size'] = global_train_batch_size
+    if microbatch_size is not None:
+        parameters['device_train_microbatch_size'] = microbatch_size
+
+    # update eval batch size based on change in seq len
+    parameters['device_eval_batch_size'] = max(1, int(parameters['device_eval_batch_size'] / ((max_seq_len / 2048) ** 2)))
+
+    parameters['train_loader']['dataset']['split'] = 'val'  # for throughput testing purposess
+    parameters['eval_loader']['eval_subset_num_batches'] = 2  # for throughput testing purposes
+
+    parameters['max_duration'] = max_duration
+    parameters['eval_interval'] = eval_interval
+
+    parameters['precision'] = precision
+    parameters['fsdp_config']['mixed_precision'] = fsdp_config_mixed_precision
+
+    if wandb:
+        # add wandb
+        parameters['loggers'] =  {'wandb': {}}
+
+    return parameters
+
+
+def get_integrations(project, wandb=True):
+    integrations = [{
+        'integration_type': 'git_repo',
+        'git_repo': 'vchiley/benchmarks',
+        'git_branch': 'llm-thru-bauto',
```

Comment:
Ideally this should work?
```suggestion
        'git_repo': 'mosaicml/examples',
        'git_branch': 'main',
```